sdks/java/core/src/main/java/com/google/cloud/dataflow/sdk/io/BoundedReadFromUnboundedSource.java

```diff
@@ -107,6 +108,21 @@ public String getKindString() {
     return "Read(" + approximateSimpleName(source.getClass()) + ")";
   }
 
+  @Override
+  public void populateDisplayData(DisplayData.Builder builder) {
+    builder
+        .add("source", source.getClass())
+        .include(source);
+
+    if (maxNumRecords != Long.MAX_VALUE) {
+      builder.add("maxRecords", maxNumRecords);
```

Comment:
We added support for some supplying a friendly key "label":

    builder.add("maxRecords", maxNumRecords)
        .withLabel("Maximum number of records returned");

.. but so far I haven't added them for any transforms. I think it's worth doing, but I'd like to discuss with @bjchambers & @ajamato first. I think consistency is desirable-- most/all transforms should have English labels, or few/none should. 

Opened [BEAM-180](https://issues.apache.org/jira/browse/BEAM-180) to track.